### PR TITLE
make typecheck required

### PR DESCRIPTION
**Pull request by BenTheElder**
Per lazy consensus, xref: #7206 
/hold
we'll have to redeploy the mungers, and splice

### CODE REVIEW

**Comment by BenTheElder** [comment]
from the verify-govet failure:
```
Cloning into 'test-infra'...
fatal: unable to access 'https://github.com/kubernetes/test-infra/': Could not resolve host: github.com
```
We should definitely not checkout test-infra in every job once we have the podutils, but that's somewhat worrying with spurious DNS failures again.

File: prow/config.yaml [hunk 1]
```diff
@@ -3808,6 +3861,59 @@ presubmits:
     max_concurrency: 0
     name: pull-security-kubernetes-cross
```
Comment (cjwagner) [inline prow/config.yaml]: This job looks identical to the one that follows besides the branch sharding?

Comment (BenTheElder) [reply prow/config.yaml]: the `run_if_changed` regex is wider for older banches (it's the existing regex which matches some go files, we test this with typecheck on master / 1.10+ now.

**Comment by BenTheElder** [comment]
/hold cancel
will deploy this now
FYI @rmmh

**Pull request merged**